### ISSUE TITLE
Improve Developer Readme

### ISSUE THREAD

**Issue by jmercouris** [atlas-engineer/nyxt#3068]
In short, do we want to:

    Seriously refactor the nix section - mention that git cloning with submodules is required, etc. I think you'll agree with that, as the document stands now, few will be able to hack on Nyxt with nix.
    Drop the section all together. I'm not saying that we should get rid of nyxt/build-scripts/shell.nix. Simply saying this hacking method will not have the "official" status. And I'm not trying to convert you to guixanism either :)

Section "Standard developer's installation" feels a bit odd in the sense that it repeats much of the installation process mentioned in the INSTALL file. Maybe it's purpose is to serve as a more in-depth reference that complements INSTALL and nyxt/documents/SOURCES.org. I think it goes too much in depth nonetheless. For instance, when it describes how to install dependencies such as webkitgtk in different GNU/Linus distributions. Package maintainers need to know the build of materials, and the rest they can figure out.

I think we should erase all references to quicklisp. We already mention that relying on it is discouraged. The git submodules render quicklisp useless. Since it's a tool that CL programmers tend to use, I think a single note somewhere mentioning why we don't rely on it suffices.

Structure-wise, the first-level contents should be (in order):

    Developer's installation guide
    Developer's environment guide (what we now call "Hacking")
    Merge "Contributing" and "Help & Community"

I suggest getting rid of "Run Nyxt in a security sandbox". nyxt.scm already mentions how to run in a container already. Does anyone actually use Firejail?

**Comment by hgluka** [comment]
Just to add on to this, if we end up removing sections like the Nix installation guide, security sandbox, and shortening others, that could make room to expand on things in other places. 

Particularly, I'm not a huge fan of linking to `build-scripts/nyxt.scm` for installation instructions. An org file is much clearer and easier to read than a comment. What do you think?

**Comment by hgluka** [comment]
You understood correctly. The `.scm` file should stay as is, but the `.org` file should also have the basic instructions and just refer to the `.scm` file for more complex use cases (e.g. running in a container).

**Comment by jmercouris** [comment]
@hgluka shall I assign you, or shall I do this?

**Comment by hgluka** [comment]
Let me just check first. The idea is this:

- [ ] Developer's installation guide (with Guix and maybe a more streamlined version of the "Standard developer's installation")
- [ ] Developer's environment guide (what I wrote in #3065)
- [ ]  Merge Contributing and Help & Community

If I'm not missing anything, then yeah, I can do it.

**Comment by jmercouris** [comment]
OK, great :-)

**Comment by aartaka** [comment]
Regarding "Contributing": this one should rather be a CONTRIBUTING file at the repository root, so that we conform to the established norms set by other projects, and so that Github suggests our contributors read this.

Moving "Programming conventions" there feels like a right move too.

**Comment by marek22k** [comment]
> I suggest getting rid of "Run Nyxt in a security sandbox". nyxt.scm already mentions how to run in a container already. Does anyone actually use Firejail?

Yes, but unfortunately [Nyxt does not work with firejail](https://github.com/netblue30/firejail/issues/1139#issuecomment-1833379156) :-(
